### PR TITLE
[CI] Update workflow to get the matrix from json file

### PR DESCRIPTION
**Pull request by zakkak**

### CODE REVIEW

**Comment by zakkak** [comment]
Blocked by https://github.com/AdoptOpenJDK/openjdk-api-v3/issues/319

**Comment by jerboaa** [comment]
@zakkak Are those test failures expected?

**Comment by zakkak** [comment]
Nope. I forgot to add `quarkus-name` to the matrix :( 

PS: You were faster than the email notification :D

**Comment by zakkak** [comment]
@jerboaa this is now finally ready for review, sorry for the early review request.

**Comment by zakkak** [comment]
@jerboaa @Karm could one of you please review this?
The failures are expected/known.

**Review by jerboaa — APPROVED** [review]
OK with me.

File: .github/workflows/mandrel.yml [hunk 1]
```diff
@@ -28,26 +28,22 @@ env:
   DB_USER: hibernate_orm_test
   DB_PASSWORD: hibernate_orm_test
   DB_NAME: hibernate_orm_test
-  NATIVE_TEST_MAVEN_OPTS: "-B --settings .github/mvn-settings.xml --fail-at-end -Dquarkus.native.additional-build-args=-J-ea -Dtest-postgresql -Dtest-elasticsearch -Delasticsearch.hosts='localhost:9200' -Dtest-keycloak -Dtest-amazon-services -Dtest-db2 -Dtest-mysql -Dtest-mariadb -Dmariadb.base_url='jdbc:mariadb://localhost:3308' -Dmariadb.url='jdbc:mariadb://localhost:3308/hibernate_orm_test'  -Dtest-mssql -Dtest-vault -Dtest-neo4j -Dtest-kafka -Dtest-redis -Dnative-image.xmx=5g -Dnative -Dnative.surefire.skip -Dformat.skip -Dno-descriptor-tests install -DskipDocs"
```
Comment (zakkak) [inline .github/workflows/mandrel.yml:31]: @Karm @jerboaa Please note the removal of `-Dquarkus.native.additional-build-args=-J-ea` this means that we no longer test with assertions enabled. 
Unfortunately defining this property overrides any possibly existing definitions (e.g. https://github.com/quarkusio/quarkus/blob/20e6bdb1c1abc11a1f4e90b43c429072a66bc9c4/integration-tests/vertx-http/src/main/resources/application.properties#L11) breaking some tests.

Sorry for not mentioning it earlier.

**Pull request merged**